### PR TITLE
Update package:vm_service dependency to ^7.0.0

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

**Comment by bkonyi** [comment]
Trying to roll through this major revision change to Flutter, so please let me know if you want me to prep for publishing in this PR as well.

File: pkgs/test_core/pubspec.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ dependencies:
   source_span: ^1.8.0
   stack_trace: ^1.10.0
   stream_channel: ^2.1.0
-  vm_service: ^6.0.0
+  vm_service: ^7.0.0
```
Comment (kevmoo) [inline pkgs/test_core/pubspec.yaml:29]: perhaps leave it as a range?

**Comment by jakemac53** [comment]
It looks like coverage also depends on it - as Kevin said if we can leave this as a range that would be ideal.

But we should also at least do a round of testing with overrides in place forcing the latest version to ensure it actually works.

**Comment by bkonyi** [comment]
> It looks like coverage also depends on it - as Kevin said if we can leave this as a range that would be ideal.

Sure, that's fine with me. However, it looks like this was a range previously. Any idea why that was changed?

> But we should also at least do a round of testing with overrides in place forcing the latest version to ensure it actually works.

I'm happy to test it, but what would that testing look like exactly?

**Comment by natebosch** [comment]
> However, it looks like this was a range previously. Any idea why that was changed?

We needed to set the minimum to the null safe version in order to do the migration in this package.

> I'm happy to test it, but what would that testing look like exactly?

```yaml
dependencies:
  vm_service: ">=6.0.0 <8.0.0"
  
dependency_overrides:
  vm_service: 7.0.0
```

After CI passes we can delete `dependency_overrides`.

Note that the overrides probably need to be put in multiple packages to get full testing in CI.

File: pkgs/test_core/pubspec.yaml [hunk 2]
```diff
@@ -26,7 +26,7 @@ dependencies:
   source_span: ^1.8.0
   stack_trace: ^1.10.0
   stream_channel: ^2.1.0
-  vm_service: ^6.0.0
+  vm_service: >= 6.0.0 < 8.0.0
```
Comment (natebosch) [inline pkgs/test_core/pubspec.yaml:29]:
```suggestion
  vm_service: >= 6.0.0 <8.0.0
```

File: pkgs/test_core/pubspec.yaml (hunk 1 above)
Comment (bkonyi) [reply pkgs/test_core/pubspec.yaml]: Done.

**Comment by bkonyi** [comment]
Looks like we're seeing failures, but they're unrelated to the vm_service version bump.

> After CI passes we can delete `dependency_overrides`.
> 
> Note that the overrides probably need to be put in multiple packages to get full testing in CI.

Wouldn't that require us committing the dependency overrides in each package and publishing with them since we pull everything from pub for testing?

**Comment by jakemac53** [comment]
> Wouldn't that require us committing the dependency overrides in each package and publishing with them since we pull everything from pub for testing?

We wouldn't publish, this would just be for testing on CI and then you can revert the commit that adds the overrides after we confirm it works.

**Comment by kevmoo** [comment]
I restarted the jobs. Looks like we're headed towards green!

**Comment by bkonyi** [comment]
> > Wouldn't that require us committing the dependency overrides in each package and publishing with them since we pull everything from pub for testing?
> 
> We wouldn't publish, this would just be for testing on CI and then you can revert the commit that adds the overrides after we confirm it works.

I'm probably just missing something here, but what do you mean by the dependency overrides in multiple packages then? The usage of overrides to test on a single repo's CI makes sense, but if we're talking multiple packages I don't think I understand what the scope of that actually is.

**Comment by jakemac53** [comment]
> I'm probably just missing something here, but what do you mean by the dependency overrides in multiple packages then? The usage of overrides to test on a single repo's CI makes sense, but if we're talking multiple packages I don't think I understand what the scope of that actually is.

Still just this repo :). But we want to make sure _all_ the tests in the repo run with vm_service 7.0.0 to get the best coverage. That means adding the override to _all_ pubspecs in the repo.

**Comment by jakemac53** [comment]
Note however that with coverage published now we might not need overrides at all - I will restart the jobs and see if they get the latest vm service, if so we are good.

**Comment by jakemac53** [comment]
Confirmed the jobs are getting vm_service 7.0.0 now, so we don't need to do any overrides.

File: pkgs/test_core/pubspec.yaml [hunk 3]
```diff
@@ -36,3 +36,4 @@ dependencies:
 dependency_overrides:
   test_api:
     path: ../test_api
+  vm_service: 7.0.0
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/test_core/pubspec.yaml:39]: Not `^7.0.0`?

[the remark this answers is not in the document]
Comment (jakemac53) [reply pkgs/test_core/pubspec.yaml:39]: this is just the override

[the remark this answers is not in the document]
Comment (kevmoo) [reply pkgs/test_core/pubspec.yaml:39]: 🤦

**Comment by bkonyi** [comment]
> > I'm probably just missing something here, but what do you mean by the dependency overrides in multiple packages then? The usage of overrides to test on a single repo's CI makes sense, but if we're talking multiple packages I don't think I understand what the scope of that actually is.
> 
> Still just this repo :). But we want to make sure _all_ the tests in the repo run with vm_service 7.0.0 to get the best coverage. That means adding the override to _all_ pubspecs in the repo.

D'oh, of course. Darn multi-package repos... :)

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**